tests/unittests/PartitionerTest.cpp

```diff
@@ -494,29 +497,66 @@ TEST_F(PartitionerTest, SimpleHeterogeneousPartitioning) {
     // Create two backends which support different ops, then do the partition by
     // assigning the ops to the corresponding abackends.
     std::vector<Backend *> backends;
-    backends.emplace_back(&backendWithoutSub1);
-    backends.emplace_back(&backendWithoutSub2);
     backends.emplace_back(&backendWithoutMul1);
     backends.emplace_back(&backendWithoutMul2);
-    std::vector<DeviceInfo> devices = {{3072, BackendKind::CPU},
-                                       {3072, BackendKind::CPU},
+    backends.emplace_back(&backendWithoutSub1);
+    backends.emplace_back(&backendWithoutSub2);
+    std::vector<DeviceInfo> devices = {{3072, BackendKind::Interpreter},
                                        {3072, BackendKind::Interpreter},
-                                       {3072, BackendKind::Interpreter}};
+                                       {3072, BackendKind::CPU},
+                                       {3072, BackendKind::CPU}};
+    auto partitioner =
+        Partitioner(&mod_, devices, backends, /* saturateHost */ true);
+    auto err = partitioner.Partition();
+    EXPECT_FALSE(errToBool(std::move(err)));
+    DAGListTy myList = std::move(partitioner.getPartitionResult());
+    ASSERT_EQ(mod_.getFunctions().size(), 3);
+    ASSERT_EQ(myList.size(), 1);
+    ASSERT_TRUE(checkSaveNode(mod_));
+
+    for (auto &dag : myList) {
+      for (auto &node : dag.nodes) {
+        // Although the saturateHost is set true, no saturating the host in
+        // heterogeneous partiton.
+        ASSERT_EQ(node->logicalDevices.size(), 1);
+      }
+    }
+    mod_.clear();
+  }
+}
+
+/// Test assigning more than one partitions in to one device.
+TEST_F(PartitionerTest, HeterogeneousPartitioning1) {
```

Comment:
Please give this test a more meaningful name.

Reply:
You just added/modified heterogeneous partitioning tests in this PR, but this new feature could be used also with a single type of backends, right? You should test this case as well.